ivy_tests/test_ivy/test_functional/test_core/test_utility.py

```diff
@@ -11,30 +11,31 @@
 
 # all
 @given(
-    data=st.data(),
+    dtype_and_x=helpers.dtype_and_values(ivy_np.valid_float_dtypes),
+    axis=helpers.integers(min_value=-1, max_value=0),
     keepdims=st.booleans(),
-    input_dtype=st.sampled_from(ivy_np.valid_float_dtypes),
     as_variable=st.booleans(),
     with_out=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="all"),
     native_array=st.booleans(),
     container=st.booleans(),
     instance_method=st.booleans(),
 )
 def test_all(
-    data,
+    dtype_and_x,
+    axis,
     keepdims,
-    input_dtype,
     as_variable,
     with_out,
+    num_positional_args,
     native_array,
     container,
     instance_method,
     fw,
 ):
-    num_positional_args = data.draw(helpers.num_positional_args(fn_name="all"))
-    shape = data.draw(helpers.get_shape(min_num_dims=1))
-    x = data.draw(helpers.array_values(dtype=input_dtype, shape=shape))
-    axis = data.draw(helpers.get_axis(shape=shape))
+    input_dtype, x = dtype_and_x
+    if not isinstance(x, list):
+        x = [x]
```

Comment:
does this function being tested not support 0-dimensional arrays? If it does, then we should not perform this conversion. If it does not, then we should try to handle this in the `helpers.dtype_and_values` helper instead, perhaps refactoring it a bit in order to support the ability to avoid generating numbers.